website/cue/reference/components.cue

```diff
@@ -119,7 +119,11 @@ components: {
 		if Kind != "sink" {
 			// `output` documents output of the component. This is very important
 			// as it communicate which events and fields are emitted.
-			output: #Output
+			output: #OutputData
+		}
+
+		if Kind == "transform" {
+			outputs: #Outputs
```

Comment:
So I think one things that not captured here is that we can have components with separate log and metric outputs. We don't currently have any of those, but while we're adjusting this model it probably makes sense to account for them.

I'm not super familiar with cue, but it seems like it might make sense to combine `output` and `outputs` into one field that captures the name and type of each of a potential list of outputs. Or maybe make it an either/or situation so that the default case of a single unnamed output is simple.

Reply:
Gotcha, if I'm understanding this correctly, it sounds like we essentially want a better way to associate `OutputData` with an `Output` (as also mentioned [here](https://github.com/vectordotdev/vector/pull/10616#discussion_r776375279))? Some way of combining `output` and `outputs` wherein each output specifies its output data makes sense to me. I'll play around with it.